### PR TITLE
[incubator/mysqlha] Use volumeClaimTemplates.spec.storageClassName

### PR DESCRIPTION
**Pull request by holmsten**
The `volume.alpha.kubernetes.io/storage-class` annotation seems to be removed(?) in Kubernetes 1.9.

After changing the template to use `volumeClaimTemplates.spec.storageClassName`, the defined storage class is correctly used and volumes are provisioned as expected.

### CODE REVIEW

**Comment by unguiculus** [comment]
Thanks for updating this. Please follow this pattern: https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md#persistence

**Comment by unguiculus** [comment]
Please have a look at the PVC example again and update accordingly.

**Comment by holmsten** [comment]
I assume you want to address:

> PVCs should support specifying an existing claim

Is it even possible to use existing PVCs in a StatefulSet? Would something like this be acceptable?

`values.yaml`:
```
persistence:
  ## Allow using existing volumes specified by label selector 
  ## ref: https://kubernetes.io/docs/concepts/storage/persistent-volumes/#selector
  ## example:
  ## 
  ## selector:
  ##   matchLabels:
  ##     foo: bar
  selector: {}
```
`templates/statefulset.yaml`
```
  volumeClaimTemplates:
  {{- if .Values.persistence.enabled }}
  - metadata:
      name: data
    spec:
      accessModes:
      - {{ .Values.persistence.accessMode | quote }}
      {{- if .Values.persistence.storageClass }}
      storageClassName: {{ .Values.persistence.storageClass }}
      {{- end }}
      resources:
        requests:
          storage: {{ .Values.persistence.size | quote }}
      selector:
{{ toYaml .Values.persistence.selector | indent 8 }}
  {{- end }}
```

**Comment by unguiculus** [comment]
Sorry, maybe I wasn't clear. I'd like you to apply the PVC example to the `volumeClaimTemplate`.

Here's an example you could use as a reference: 
https://github.com/kubernetes/charts/blob/master/stable/mongodb-replicaset/templates/mongodb-statefulset.yaml#L225-L250

**Comment by holmsten** [comment]
I've used the mongodb example as reference, is it up to standards now?

File: incubator/mysqlha/templates/statefulset.yaml [hunk 1]
```diff
@@ -223,21 +223,30 @@ spec:
           name: {{ template "fullname" . }}
       - name: scripts
         emptyDir: {}
-      {{- if eq .Values.persistence.enabled false }}
+{{- if not .Values.persistence.enabled }}
```
Comment (unguiculus) [inline incubator/mysqlha/templates/statefulset.yaml]: Don't use `not` and swap the contents of the if/else blocks.

File: incubator/mysqlha/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: mysqlha
-version: 0.1.3
+version: 0.1.4
```
Comment (unguiculus) [inline incubator/mysqlha/Chart.yaml]: Make it a 0.2.0.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by unguiculus** [comment]
cc @jpoon

**Comment by jpoon** [comment]
as storage classes were introduced in 1.6, can you update the readme to denote that k8s 1.6 is now required?

aside from that lgtm.

File: incubator/mysqlha/values.yaml [hunk 3]
```diff
@@ -34,10 +34,17 @@ mysqlha:
 ##
 persistence:
   enabled: true
-  ## If defined, volume.beta.kubernetes.io/storage-class: <storageClass>
+  ## If defined, storageClassName: <storageClass>
+  ## If set to "-", storageClassName: "", which disables dynamic provisioning
+  ## If undefined (the default) or set to null, no storageClassName spec is
+  ##   set, choosing the default provisioner.  (gp2 on AWS, standard on
+  ##   GKE, AWS & OpenStack)
```
Comment (jpoon) [inline incubator/mysqlha/values.yaml]: But...what about Azure?!

File: incubator/mysqlha/templates/statefulset.yaml (hunk 1 above)
Comment (holmsten) [reply incubator/mysqlha/templates/statefulset.yaml]: Why should not `not` be used?

File: incubator/mysqlha/values.yaml (hunk 3 above)
Comment (holmsten) [reply incubator/mysqlha/values.yaml]: I've never used Azure so I don't know what is used there. `azure-disk` perhaps - referring to https://docs.microsoft.com/en-us/azure/aks/azure-disks-dynamic-pv?

File: incubator/mysqlha/templates/statefulset.yaml (hunk 1 above)
Comment (holmsten) [reply incubator/mysqlha/templates/statefulset.yaml]: Now swapped

File: incubator/mysqlha/Chart.yaml (hunk 2 above)
Comment (holmsten) [reply incubator/mysqlha/Chart.yaml]: Done

File: incubator/mysqlha/values.yaml (hunk 3 above)
Comment (holmsten) [reply incubator/mysqlha/values.yaml]: Added documentation for default provisioner used in Azure.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**